### DISCUSSION TITLE
How to make whisper aware of my domain language ?

### DISCUSSION THREAD

**Discussion by haikyuu** [ggml-org/whisper.cpp#580] [Q&A]
I'm using whisper.cpp to transcribe coding tutorials. And the results I get look like this
```
... index dot CSS file , ... the h 1 and an h 2
```
And I want whisper to output `index.css ... h1 and h2` instead

Is that possible somehow using whisper.cpp? should I "fine tune" the model? If so, I'd appreciate any useful resources to do that.

**Comment by ggerganov** [comment]
There is no correct way to control this.

You can "influence" to some extend the transcription by providing a `--prompt`.
See https://github.com/ggerganov/whisper.cpp/discussions/348 and https://github.com/ggerganov/whisper.cpp/issues/393 for examples of how to use it, but not sure if it will solve your problem.

**Reply by haikyuu** [reply]
Got you. I'll check it out. Although I'm more leaning on doing that as a cleanup step after transcriptions

By the way (not related to the same topic), but Whisper just added word level timestamps to the main branch and it has an option that's a bit useful and we could probably add that to whisper.cpp https://github.com/openai/whisper/pull/869/files#diff-f6accbbb4ebcd3dd6815bf012490d9ba37eb89a65f2124adc95c2a39bc6941b7R340
```
parser.add_argument("--append_punctuations", type=str, default="\"\'.。,，!！?？:：”)]}、", help="if word_timestamps is True, merge these punctuation symbols with the previous word")
```

**Reply by ggerganov** [reply]
Sure, open an issue for this

**Comment by elijahfhopp** [comment]
Additionally, if the output is uniform, you may transform it after it is transcribed. You may have trouble finding when to start and end the transformation, but selecting the start word and having it parse until it errors may be workable. I know this is kinda an obvious and dumb solution, but I thought I would mention it just for.

Edit: I missed your above comment, :/.
